pkg/api/aim2/dao/repositories/experiment.go

```diff
@@ -44,149 +52,247 @@ func NewExperimentRepository(db *gorm.DB) *ExperimentRepository {
-func (r ExperimentRepository) GetByNamespaceIDAndExperimentID(
-	ctx context.Context, namespaceID uint, experimentID int32,
-) (*models.Experiment, error) {
-	var experiment models.Experiment
-	if err := r.db.WithContext(ctx).Preload(
-		"Tags",
-	).Where(
-		models.Experiment{ID: &experimentID},
-	).Where(
-		"experiments.namespace_id = ?", namespaceID,
-	).First(&experiment).Error; err != nil {
-		return nil, eris.Wrapf(err, "error getting experiment by id: %d", experimentID)
-	}
-	return &experiment, nil
-}
-
-// GetByNamespaceIDAndName returns experiment by Namespace ID and Experiment name.
-func (r ExperimentRepository) GetByNamespaceIDAndName(
-	ctx context.Context, namespaceID uint, name string,
-) (*models.Experiment, error) {
-	var experiment models.Experiment
-	if err := r.db.WithContext(ctx).Preload(
-		"Tags",
-	).Where(
-		models.Experiment{Name: name},
-	).Where(
-		"experiments.namespace_id = ?", namespaceID,
-	).First(&experiment).Error; err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return nil, nil
-		}
-		return nil, eris.Wrapf(err, "error getting experiment by id: %s", name)
-	}
-	return &experiment, nil
-}
-
-// Update updates existing models.Experiment entity.
+// Update updates existing experiment.
 func (r ExperimentRepository) Update(ctx context.Context, experiment *models.Experiment) error {
-	if err := r.db.Transaction(func(tx *gorm.DB) error {
+	return r.db.Transaction(func(tx *gorm.DB) error {
 		if err := tx.WithContext(ctx).Model(&experiment).Updates(experiment).Error; err != nil {
 			return eris.Wrapf(err, "error updating experiment with id: %d", *experiment.ID)
 		}
 
 		// also archive experiment runs if experiment is being archived
 		if experiment.LifecycleStage == models.LifecycleStageDeleted {
-			run := models.Run{
-				LifecycleStage: experiment.LifecycleStage,
-				DeletedTime:    experiment.LastUpdateTime,
-			}
-
 			if err := tx.WithContext(
 				ctx,
 			).Model(
-				&run,
+				&models.Run{},
 			).Where(
 				"experiment_id = ?", experiment.ID,
-			).Updates(&run).Error; err != nil {
+			).Updates(&models.Run{
+				LifecycleStage: experiment.LifecycleStage,
+				DeletedTime:    experiment.LastUpdateTime,
+			}).Error; err != nil {
 				return eris.Wrapf(err, "error updating existing runs with experiment id: %d", *experiment.ID)
 			}
 		}
 		return nil
-	}); err != nil {
-		return err
-	}
-
-	return nil
+	})
 }
 
-// Delete removes the existing models.Experiment from the db.
+// Delete deletes existing experiment.
 func (r ExperimentRepository) Delete(ctx context.Context, experiment *models.Experiment) error {
-	return r.DeleteBatch(ctx, []*int32{experiment.ID})
-}
-
-// DeleteBatch removes existing []models.Experiment in batch from the db.
-func (r ExperimentRepository) DeleteBatch(ctx context.Context, ids []*int32) error {
 	if err := r.db.Transaction(func(tx *gorm.DB) error {
-		// finding all the runs
+		// finding all the related runs.
```

Comment:
i don't think short comments should have closing period unless they are actually sentences (this is not)